File: RCTNtlDocumentCamera/RCTNtlDocumentCamera.m
//
//  RCTNtlDocumentCamera.m
//  RCTNtlDocumentCamera
//
//  Created by Breeshy Sama on 22/10/2563 BE.
//

#import "RCTNtlDocumentCamera.h"



@implementation RCTNtlDocumentCamera

- (dispatch_queue_t)methodQueue
{
    return dispatch_get_main_queue();
}

RCT_EXPORT_MODULE();

- (NSArray<NSString *> *)supportedEvents
{
    return @[@"EventReminder"];
}

RCT_EXPORT_METHOD(openCamera)
{
    [self configueDocumentView];
};


-(void)configueDocumentView
{
    VNDocumentCameraViewController* documentCameraViewController = [[VNDocumentCameraViewController alloc] init];
    
    documentCameraViewController.delegate = self;
    
    [[UIApplication sharedApplication].delegate.window.rootViewController presentViewController:documentCameraViewController animated:YES completion:nil];
}

- (void)documentCameraViewController:(VNDocumentCameraViewController *)controller didFinishWithScan:(VNDocumentCameraScan *)scan{
    NSMutableArray *imageArray = [NSMutableArray array];
    int i;
    for (i = 0; i < [scan pageCount]; i++) {
        // do something with object
        UIImage *image =  [scan imageOfPageAtIndex:i];
        NSData *imageData = UIImagePNGRepresentation(image);
        NSString * base64String = [imageData base64EncodedStringWithOptions:0];
        [imageArray addObject:base64String];
        //        NSLog (@"image = %@", image);
    }
    NSLog (@"images = %@", imageArray);
    [self sendEventWithName:@"EventReminder" body:@{@"images":imageArray}];
    [controller dismissViewControllerAnimated:true completion:nil];
}

@end

File: RCTNtlDocumentCamera/RCTNtlDocumentCamera.swift
import UIKit
import  VisionKit


@objc(NtlDocumentCameraHelper)
class NtlDocumentCameraHelper: UIViewController {
    private func configueDocumentView(){
        let scanningDocumentVC = VNDocumentCameraViewController()
        scanningDocumentVC.delegate = self
        self.present(scanningDocumentVC, animated: true, completion:nil)
    }
}
 
extension NtlDocumentCameraHelper:VNDocumentCameraViewControllerDelegate{
    func documentCameraViewController(_ controller:VNDocumentCameraViewController,didFinishWith scan:VNDocumentCameraScan) {
        for pageNumber in 0..<scan.pageCount{
            let image = scan.imageOfPage(at: pageNumber)
            print(image)
        }

        controller.dismiss(animated: true, completion: nil)
    }
}
